src/textual/css/styles.py

```diff
@@ -295,6 +333,71 @@ def get_transition(self, key: str) -> Transition | None:
         else:
             return None
 
+    def get_box_model(
+        self, container_size: Size, parent_size: Size
+    ) -> tuple[Size, Spacing]:
+        """Resolve the box model for this Styles.
+
+        Args:
+            parent_size (Size): The size of the widget container.
+            terminal_size (Size): The size of the terminal.
+
+        Returns:
+            tuple[Size, Spacing]: A tuple with the size of the content area and margin.
+        """
+        has_rule = self.has_rule
+        width, height = container_size
+
+        if has_rule("width"):
+            width = self.width.resolve_dimension(container_size, parent_size)
+        else:
+            width = max(0, width - self.margin.width)
+
+        if self.min_width:
+            min_width = self.min_width.resolve_dimension(container_size, parent_size)
+            width = max(width, min_width)
+
+        if self.max_width:
+            max_width = self.max_width.resolve_dimension(container_size, parent_size)
+            width = min(width, max_width)
+
+        if has_rule("height"):
+            height = self.height.resolve_dimension(container_size, parent_size)
+        else:
+            height = max(0, height - self.margin.height)
+
+        if self.min_height:
+            min_height = self.min_height.resolve_dimension(container_size, parent_size)
+            height = max(height, min_height)
+
+        if self.max_height:
+            max_height = self.max_height.resolve_dimension(container_size, parent_size)
+            height = min(width, max_height)
+
+        # TODO: box sizing
+
+        size = Size(width, height)
+        margin = Spacing(0, 0, 0, 0)
+
+        if self.box_sizing == "content-box":
```

Comment:
This differs from the CSS box model and feels very surprising to me - "content-box" shouldn't include padding or border in the size, it's only the size of the content box itself. This will cause confusion going forward if we implement CSS rules with different semantics from CSS itself.

Reply:
I think that is correct. The `size` being calculated includes is the space taken up by content + padding + border.